app/v2/controllers/V2DeparturesController.scala

```diff
@@ -70,26 +77,56 @@ class V2DeparturesControllerImpl @Inject() (
   def submitDeclaration(): Action[Source[ByteString, _]] =
     (authActionNewEnrolmentOnly andThen messageSizeAction()).async(streamFromMemory) {
       implicit request =>
-        withTemporaryFile {
-          (temporaryFile, source) =>
-            logger.info("Version 2 of endpoint has been called")
-
-            implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-
-            (for {
-              _ <- validationService.validateXML(MessageType.DepartureDeclaration, source).asPresentation
-
-              fileSource = FileIO.fromPath(temporaryFile)
-
-              declarationResult <- departuresService.saveDeclaration(request.eoriNumber, fileSource).asPresentation
-              _ <- routerService
-                .send(MessageType.DepartureDeclaration, request.eoriNumber, declarationResult.departureId, declarationResult.messageId, fileSource)
-                .asPresentation
-            } yield declarationResult).fold[Result](
-              presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-              result => Accepted(HateoasDepartureDeclarationResponse(result.departureId))
-            )
-        }.toResult
+        logger.info("Version 2 of endpoint has been called")
+
+        request.headers.get(HeaderNames.CONTENT_TYPE) match {
```

Comment:
Not sure about this, but could there be an issue with case sensitivity?

Reply:
No, just looked, Play uses a `CaseInsensitiveOrdered` `TreeMap`, so everything is case insensitive here.

Reply:
(for the curious, see `play.api.mvc.Headers#toMap`)

Reply:
Cool, thanks @drnaylor-oc